tests/unittests/test_core/test_archiver.py

```diff
@@ -32,127 +37,323 @@ def setup_class(cls):
         )
         os.mkdir(cls.tmp_path)
 
-        # make LDF
-        os.mkdir(os.path.join(cls.tmp_path, "images"))
-        cls.ldf_name = "dummyLDF"
-        labels = ["label1", "label2", "label3"]
+        # make LDFs
+        unilabelLDF = "dummyLDF_unilabel"
+        cls._make_dummy_ldf(
+            ldf_name=unilabelLDF,
+            save_path=cls.tmp_path,
+            bbx_anno=True,
+            kpt_anno=True,
+        )
+        multilabelLDF = "dummyLDF_multilabel"
+        cls._make_dummy_ldf(
+            ldf_name=multilabelLDF,
+            save_path=cls.tmp_path,
+            cls_anno=True,
+            bbx_anno=True,
+            sgm_anno=True,
+            multilabel=True,
+        )
+        cls.ldf_names = [unilabelLDF, multilabelLDF]
+
+        for head_name in HEAD_NAMES:
+            if head_name == "ImplicitKeypointBBoxHead":
+                ldf_name = unilabelLDF  # multiclass keypoint detection not yet supported in luxonis-train
+            else:
+                ldf_name = multilabelLDF
+
+            # make config
+            cfg_dict = cls._make_dummy_cfg_dict(
+                head_name=head_name,
+                save_path=cls.tmp_path,
+                ldf_name=ldf_name,
+            )
+            cfg = Config.get_config(cfg_dict)
+
+            # train model
+            cfg.trainer.epochs = 1
+            cfg.trainer.validation_interval = 1
+            cfg.trainer.batch_size = 1
+            trainer = Trainer(cfg=cfg)
+            trainer.train()
+            callbacks = [
+                c
+                for c in trainer.pl_trainer.callbacks
+                if isinstance(c, pl.callbacks.ModelCheckpoint)
+            ]
+            model_checkpoint_path = callbacks[0].best_model_path
+            model_ckpt = os.path.join(trainer.run_save_dir, model_checkpoint_path)
+            trainer.reset_logging()
+
+            # export model to ONNX
+            cfg.model.weights = model_ckpt
+            exporter = Exporter(cfg=cfg)
+            cls.onnx_model_path = os.path.join(cls.tmp_path, "model.onnx")
+            exporter.export(onnx_path=cls.onnx_model_path)
+            exporter.reset_logging()
+
+            # make archive
+            cfg.archiver.archive_save_directory = cls.tmp_path
+            cfg.archiver.archive_name = f"nnarchive_{head_name}"
+            archiver = Archiver(cfg=cfg)
+            cls.archive_path = archiver.archive(cls.onnx_model_path)
+            archiver.reset_logging()
+
+            # clear the loaded config instance
+            Config.clear_instance()
 
-        def classification_dataset_generator():
-            for i in range(10):
-                img = np.random.randint(0, 256, (10, 10, 3), dtype=np.uint8)
-                img_file_path = os.path.join(cls.tmp_path, "images", f"img{i}.png")
+    def _make_dummy_ldf(
```

Comment:
Not important and just an idea, but I could see this being useful across other tests too. We could make a `LuxonisDatasetFactory` as a shared factory which can be used in other tests as well.

Reply:
It might also make sense to instead implement it in `luxonis-ml` and import it here since it would be useful for `luxonis-ml` tests

Reply:
Sure, it makes sense to move it to luxonis-ml!